BlobStorage/Advanced.cs

```diff
@@ -1431,52 +1182,42 @@ private static async Task CreateSequentiallyNamedBlockBlobsAsync(CloudBlobContai
         /// <param name="numberOfBlobsPerLevel">The number of blobs to create per level.</param>
         /// <returns>A Task object.</returns>
-        private static async Task CreateNestedBlockBlobsAsync(CloudBlobContainer container, short numberOfLevels, short numberOfBlobsPerLevel)
+        private static async Task CreateNestedBlockBlobsAsync(BlobContainerClient container, short numberOfLevels, short numberOfBlobsPerLevel)
         {
             try
             {
-                CloudBlockBlob blob;
-                MemoryStream msWrite;
+                BlobClient blob;
                 string blobName = string.Empty;
                 string virtualDirName = string.Empty;
 
                 // Create a nested blob structure.
                 for (int i = 1; i <= numberOfLevels; i++)
                 {
                     // Construct the virtual directory name, which becomes part of the blob name.
-                    virtualDirName += string.Format("level{0}{1}", i, container.ServiceClient.DefaultDelimiter);
+                    virtualDirName += string.Format("level{0}", i);
                     for (int j = 1; j <= numberOfBlobsPerLevel; j++)
                     {
                         // Construct string for blob name.
                         blobName = virtualDirName + string.Format("{0}-{1}.txt", i, j.ToString("00000"));
 
                         // Get a reference to the blob.
-                        blob = container.GetBlockBlobReference(blobName);
+                        blob = container.GetBlobClient(blobName);
 
                         // Write the blob URI to its contents.
-                        msWrite = new MemoryStream(Encoding.UTF8.GetBytes("Absolute URI to blob: " + blob.StorageUri.PrimaryUri + "."));
-                        msWrite.Position = 0;
-                        using (msWrite)
-                        {
-                            await blob.UploadFromStreamAsync(msWrite);
-                        }
-
-                        // Set a property on the blob.
-                        blob.Properties.ContentType = "text/html";
-                        await blob.SetPropertiesAsync();
-
+                        await blob.UploadAsync(BinaryData.FromString($"Absolute URI to blob: " + blob.Uri.AbsoluteUri + "."));
+                        Dictionary<string, string> metadata = new Dictionary<string, string>();
                         // Set some metadata on the blob.
-                        blob.Metadata.Add("DateCreated", DateTime.UtcNow.ToLongDateString());
-                        blob.Metadata.Add("TimeCreated", DateTime.UtcNow.ToLongTimeString());
-                        await blob.SetMetadataAsync();
+                        metadata.Add("DateCreated", DateTime.UtcNow.ToLongDateString());
+                        metadata.Add("TimeCreated", DateTime.UtcNow.ToLongTimeString());
+                        await blob.SetMetadataAsync(metadata);
```

Comment:
Again we can set the metadata in the same service call as the upload. Simpler code and cheaper for customers.